Add support for environment.yml file

martinRenou opened the pull request:
Fix https://github.com/jupyterlite/xeus-python-kernel/issues/53

- [x] Add tests (Run the tests from the test directory and put an `environment.yml` file there)
- [x] Make the "environment.yml" file name configurable
- [x] Update docs
- [x] Update README

nthiery commented:
Nice! I did not expect it to come so soon; thank you @martinRenou. Eager to try it.

Do you have some tentative timeline for when pip support might be added?

martinRenou commented:
> Do you have some tentative timeline for when pip support might be added?

Probably pure Python packages could simply be installed with `pip` and it might just be a matter of removing the check I added [here](https://github.com/jupyterlite/xeus-python-kernel/pull/96/files#diff-c9a50c38e06780356510caad3d24a845916862fdacbdba94540f718301c01365R163). I would need to give it a try :)

I don't see any reason for pure Python packages to not work if installed with `pip`, I added the check because I'm not sure how to handle the non-pure packages, how to track them and show a nice error message at build time. Maybe we could just let it fail at runtime, though it would probably prevent the kernel to start without any feedback to the user.
